test/integration/gardenlet/backupbucket/backupbucket_test.go

```diff
@@ -0,0 +1,285 @@
+// limitations under the License.
+
+package backupbucket_test
+
+import (
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("BackupBucket controller tests", func() {
+	var (
+		backupBucket                      *gardencorev1beta1.BackupBucket
+		generatedSecret                   *corev1.Secret
+		coreGeneratedSecret               *corev1.Secret
+		extensionSecret                   *corev1.Secret
+		generatedSecretRef                *corev1.SecretReference
+		coreGeneratedSecretRef            *corev1.SecretReference
+		extensionBackupBucket             *extensionsv1alpha1.BackupBucket
+		expectedExtensionBackupBucketSpec extensionsv1alpha1.BackupBucketSpec
+
+		backupBucketReady = func(makeReady bool) {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(extensionSecret), extensionSecret)).To(Succeed())
+				g.Expect(extensionSecret.Data).To(Equal(secret.Data))
+
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupBucket), extensionBackupBucket)).To(Succeed())
+				g.Expect(extensionBackupBucket.Spec).To(Equal(expectedExtensionBackupBucketSpec))
+				g.Expect(extensionBackupBucket.Annotations).To(HaveKeyWithValue(v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile))
+			}).Should(Succeed())
+
+			Expect(testClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupBucket), extensionBackupBucket)).To(Succeed())
+			patch := client.MergeFrom(extensionBackupBucket.DeepCopy())
+			delete(extensionBackupBucket.Annotations, v1beta1constants.GardenerOperation)
+			Expect(testClient.Patch(ctx, extensionBackupBucket, patch)).To(Succeed())
+
+			// These should be done by the extension controller, we are faking it here for the tests.
+			Expect(testClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupBucket), extensionBackupBucket)).To(Succeed())
+			patch = client.MergeFrom(extensionBackupBucket.DeepCopy())
+
+			lastOperationState := gardencorev1beta1.LastOperationStateSucceeded
+			if !makeReady {
+				lastOperationState = gardencorev1beta1.LastOperationStateError
+			}
+			extensionBackupBucket.Status = extensionsv1alpha1.BackupBucketStatus{
+				DefaultStatus: extensionsv1alpha1.DefaultStatus{
+					ObservedGeneration: extensionBackupBucket.Generation,
+					ProviderStatus:     &runtime.RawExtension{Raw: []byte("{\"foo\":\"bar\"}")},
```

Comment:
You could use backticks to prevent escaping the `"`


Also, probably this content should be stored in a variable and re-used below?